### ISSUE TITLE
Unit test strError.returnSuccessAfterClosingFile fails on Fedora

### ISSUE THREAD

**Issue by D4N** [Exiv2/exiv2#196]
When working on #194 I discovered that the unit test `strError.returnSuccessAfterClosingFile` fails on Fedora 26 & 27:
```
[ RUN      ] strError.returnSuccessAfterClosingFile
/home/dan/projects/exiv2/unitTests/test_futils.cpp:24: Failure
      Expected: expectedString
      Which is: "Success (errno = 0)"
To be equal to: strError().c_str()
      Which is: "No such file or directory (errno = 2)"
[  FAILED  ] strError.returnSuccessAfterClosingFile (1 ms)
```
this happens as well with gcc as with clang, but I suspect that the difference in behavior is due to glibc, which is at version 2.25 and 2.26 for Fedora 26 and 27, respectively.

@piponazo Do you have a guess what the problem could be?

**Comment by piponazo** [comment]
Hum, that's interesting. I have never dug into glibc details to be able to answer to that question. 

The test is creating a temporary empty file (with std::ofstream) and then closing it. After that we call the strError function just to check if the action of closing the file was a success. I would check the following things:

- Is the file really created ? Try to comment the line 27 of that test (`std::remove`) and check afterwards if the file is in the directory from where you are calling the test runner.
- You could also try to write something into the stream to be sure that the file is created. 
- Maybe you are running the test runner from a directory where your user has not rights to write there ?

**Comment by D4N** [comment]
The file is being created and I can write it too, so thats not a problem. I have found out that `errno` is already set before the temporary file is created. So I have re-run the unit tests with strace and it turns out that SELinux is running a some syscalls, among them the following:
```
openat(AT_FDCWD, "/etc/pki/tls/legacy-settings", O_RDONLY) = -1 ENOENT (No such file or directory)
```
I have checked the C documentation of `errno` and the issue here is, that `errno` is only guaranteed to be set when a system call fails. It can be set on succes but it does not need to be (and apparently the systemcalls after the `openat` do not reset `errno`).

The fix for that is relatively simple: just set `errno` to zero before the test. I'll make a PR for this.
